delfin/drivers/hitachi/vsp/vsp_stor.py

```diff
@@ -0,0 +1,257 @@
+            capacity_json = self.rest_handler.get_capacity()
+            free_total = capacity_json.get("total").get("freeSpace") * units.Ki
+            total_total = capacity_json.get("total").get("totalCapacity") * \
+                units.Ki
+            used_total = total_total - free_total
+        else:
+            pools_info = self.rest_handler.get_all_pools()
+            if pools_info is not None:
+                pools = pools_info.get('data')
+                for pool in pools:
+                    total_cap = \
+                        int(pool.get(
+                            'totalPoolCapacity')) * units.Mi
+                    free_cap = int(
+                        pool.get(
+                            'availableVolumeCapacity')) * units.Mi
+                    free_total = free_total + free_cap
+                    total_total = total_total + total_cap
+                used_total = total_total - free_total
+        firmware_version = self.rest_handler.get_storage()
+        status = constants.StorageStatus.OFFLINE
+        if firmware_version is not None:
+            status = constants.StorageStatus.NORMAL
+        system_name = '%s_%s' % (self.rest_handler.device_model,
+                                 self.rest_handler.rest_client.rest_host)
+
+        s = {
+            'name': system_name,
+            'vendor': 'Hitachi',
+            'description': 'Hitachi VSP Storage',
+            'model': str(self.rest_handler.device_model),
+            'status': status,
+            'serial_number': str(self.rest_handler.serial_number),
+            'firmware_version': str(firmware_version),
+            'location': '',
+            'raw_capacity': int(total_total),
+            'total_capacity': int(total_total),
+            'used_capacity': int(used_total),
+            'free_capacity': int(free_total)
+        }
+        return s
+
+    def list_storage_pools(self, context):
+        try:
+            pools_info = self.rest_handler.get_all_pools()
+            pool_list = []
+            pools = pools_info.get('data')
+            for pool in pools:
+                status = self.POOL_STATUS_MAP.get(
+                    pool.get('poolStatus'),
+                    constants.StoragePoolStatus.ABNORMAL
+                )
+                storage_type = constants.StorageType.BLOCK
+                total_cap = \
+                    int(pool.get('totalPoolCapacity')) * units.Mi
+                free_cap = int(
+                    pool.get('availableVolumeCapacity')) * units.Mi
+                used_cap = total_cap - free_cap
+                p = {
+                    'name': pool.get('poolName'),
+                    'storage_id': self.storage_id,
+                    'native_storage_pool_id': str(pool.get('poolId')),
+                    'description': 'Hitachi VSP Pool',
+                    'status': status,
+                    'storage_type': storage_type,
+                    'subscribed_capacity': int(total_cap),
```

Comment:
remove subscribed_capacity if there is no subscribed capacity

Reply:
if remove subscribed_capacity ,it will failed in DME